libs/header/p2p/metrics.go

```diff
@@ -0,0 +1,54 @@
+package p2p
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncfloat64"
+)
+
+type metrics struct {
+	responseSize     syncfloat64.Histogram
+	responseDuration syncfloat64.Histogram
+}
+
+var (
+	meter = global.MeterProvider().Meter("header/p2p")
+)
+
+func (ex *Exchange[H]) RegisterMetrics() error {
+	responseSize, err := meter.
+		SyncFloat64().
+		Histogram(
+			"header_p2p_headers_response_size",
+			instrument.WithDescription("Size of get headers response in bytes"),
+		)
+	if err != nil {
+		return err
+	}
+
+	responseDuration, err := meter.
+		SyncFloat64().
+		Histogram(
+			"header_p2p_headers_request_duration",
+			instrument.WithDescription("Duration of get headers request in seconds"),
+		)
+	if err != nil {
+		return err
+	}
+
+	ex.metrics = &metrics{
+		responseSize:     responseSize,
+		responseDuration: responseDuration,
+	}
+	return nil
+}
+
+func (m *metrics) ObserveRequest(ctx context.Context, size uint64, duration uint64) {
```

Comment:
does this have to be exported?

Reply:
also we should call this `observeResponse`

Reply:
and IMO it should also record the err if there is one

Reply:
Agree with Response and error tracking, don't really care about publicity if it's anyway inaccessible

Reply:
still weird pattern, lets privatise.

Reply:
Good call, will implemented